File: src/Row with max 1s/Code/solution.py
class Solution:
    def rowWithMax1s(self, arr):
        n = len(arr)
        m = len(arr[0])
        
        max_ones = 0
        answer = -1
        
        for i in range(n):
            low, high = 0, m - 1
            first_one = -1
            
            # Binary search for first 1
            while low <= high:
                mid = (low + high) // 2
                
                if arr[i][mid] == 1:
                    first_one = mid
                    high = mid - 1
                else:
                    low = mid + 1
            
            if first_one != -1:
                ones_count = m - first_one
                
                if ones_count > max_ones:
                    max_ones = ones_count
                    answer = i
        
        return answer

File: src/Row with max 1s/Code/solution.cpp
class Solution
{
public:
    int rowWithMax1s(vector<vector<int>> &arr)
    {
        int n = arr.size();
        int m = arr[0].size();

        int maxOnes = 0;
        int answer = -1;

        // Traverse each row
        for (int i = 0; i < n; i++)
        {
            int low = 0, high = m - 1;
            int firstOne = -1;

            // Binary search for first 1
            while (low <= high)
            {
                int mid = low + (high - low) / 2;

                if (arr[i][mid] == 1)
                {
                    firstOne = mid;
                    high = mid - 1;
                }
                else
                {
                    low = mid + 1;
                }
            }

            // If row contains at least one 1
            if (firstOne != -1)
            {
                int onesCount = m - firstOne;

                if (onesCount > maxOnes)
                {
                    maxOnes = onesCount;
                    answer = i;
                }
            }
        }

        return answer;
    }
};
